exact_multiple_string_matching/commentz_walter.py

```diff
@@ -0,0 +1,184 @@
+from dataclasses import dataclass
+from collections import deque
+
+@dataclass
+class Node():
+  def __init__(self, char, depth, parent):
+    self.char = char
+    self.depth = depth
+    self.word = None
+    self.parent = parent
+    self.ac_suffix = None
+    self.ac_output = None
+    self.children = {}
+
+@dataclass
+class CommentzWalterNode(Node):
+  def __init__(self, char, depth, parent):
+    Node.__init__(self, char, depth, parent)
+    self.min_diff_s1 = -1
+    self.min_diff_s2 = -1
+    self.cw_suffix = None
+    self.cw_output = None
+    self.shift1 = None
+    self.shift2 = None
+
+# pylint: disable=no-self-use
+class Trie():
+  def create_node(self, char, depth, parent):
+    return Node(char, depth, parent)
+
+  def __init__(self):
+    self.size = 0
+    self.root = self.create_node(None, 0, None)
+
+  def add_word(self, word):
+    curr_node = self.root
+    curr_depth = 1
+    for char in word:
+      next_node = curr_node.children.get(char)
+      if not next_node:
+        next_node = self.create_node(char, curr_depth, curr_node)
+        curr_node.children[char] = next_node
+
+      curr_node = next_node
+      curr_depth += 1
+
+    if curr_node.word is not None:
+      return
+
+    curr_node.word = word
+    self.size += 1
```

Comment:
Since these are the last lines, probably reverse is better:
if curr_node.word is None:
  curr_node.word = word
  self.size += 1